sfge/src/main/java/com/salesforce/rules/unusedmethod/RuleStateTracker.java

```diff
@@ -371,4 +409,35 @@ Optional<BaseSFVertex> getDeclarationOfReferencedValue(MethodCallExpressionVerte
         // Optional and let the caller figure out what to do with it.
         return Optional.empty();
     }
+
+    /**
+     * If an inner class shares the name of an outer class, then references to the inner class that
+     * occur in its outer/sibling classes can resemble references to the outer class. This class
+     * filters such false positives from the results of a query.
+     *
+     * @param unfilteredResults - The results, pre-filtering
+     * @param referencedType - The outer type with which inner types may collide
+     * @return - {@code unfilteredResults}, with all colliding references removed
+     */
+    private <T extends BaseSFVertex> List<T> removeInnerTypeCollisions(
```

Comment:
The else-branch from `getInvocationsOnType` and `getConstructionsOfType` are identical, and have been refactored into their own method.